### PR TITLE
feat: status and elapsed time indicator

### LINKED ISSUE #173: feat(tui): show a clear status indicator when research is finished
**Is your feature request related to a problem? Please describe.**
When a research session completes its tool-use / browsing phase and transitions to aggregating and writing the final report, there is no clear visual signal. The "Analyzing results..." spinner appears briefly, but the gap between the last visible tool activity and the final response can feel like a stall, which means that the user doesn't know if the agent is still working or if something went wrong. This leads users to prompt the agent unnecessarily, potentially disrupting the generation.

**Describe the solution you'd like**
Add explicit, distinct status messages for each major phase transition:

1. **"Researching..."**: while tools/sub-agents are actively running.
2. **"Writing report..."**: once all tools are done and the model is generating the final response. This is the phase that currently looks like a stall.
3. **"Done"**: a clear completion banner (with a timestamp or elapsed time) when the response is fully rendered.

These should be visible in the status bar (`status_bar.py`) or as inline system messages so the user always knows the current phase.

**Describe alternatives you've considered**
- **Progress bar with percentage**: hard to estimate accurately since we don't know total tool calls in advance, but a "tools completed: N" counter would help.
- **Elapsed-time timer**: show a running clock so silence doesn't feel like a hang. Lower-effort than phase labels but still helpful.
- **Audio/desktop notification on completion**: useful for long-running research where the user switches to another window. Could complement visual indicators.

**Additional context**
The streaming state machine (`stream/state.py`) already tracks `is_processing`, `is_thinking`, `is_responding`, and `is_summarizing` (lines ~104-106) and receives a `"done"` event (line ~323). The TUI's status bar (`status_bar.py`) currently only shows an MCP-loading spinner. These existing states could be surfaced with minimal wiring.

The `_is_final_response()` check combined with `not state.response_text` (tui_interactive.py, line ~1394) is the exact moment the "stall" perception begins, and this is where a more prominent "Writing report..." indicator should appear.

### PR DESCRIPTION
**Pull request by mooshroom4422**
## Description

Closes #173. Changes status bar info to `` Phase... (elapsed time) `` , adds total elapsed time information to token usage widget.

## Type of change

- [ ] Bug fix
- [x] New feature — link issue: #
- [ ] Documentation / examples
- [ ] Test improvement
- [ ] Refactor (no behavior change)

## Checklist

- [x] I have read the [Contributing Guidelines](../CONTRIBUTING.md)
- [x] This targets **core functionality** used by the majority of users (niche features belong in [EvoSkills](https://github.com/EvoScientist/EvoSkills))
- [x] I have added/updated tests where applicable
- [x] `uv run ruff check .` passes
- [x] `uv run pytest` passes

## Summary by CodeRabbit

* **New Features**
  * Status bar shows more granular phases (idle, thinking, researching, writing, done) with an updated hint color for the writing phase.
  * Token usage widget now includes per-turn elapsed time when available.

* **Improvements**
  * More accurate phase-driven status updates during interactions for clearer, real-time feedback.

[Review Change Stack](https://app.coderabbit.ai/change-stack/EvoScientist/EvoScientist/pull/218)

### CODE REVIEW

File: EvoScientist/cli/tui_interactive.py [hunk 1]
```diff
@@ -1384,6 +1382,14 @@ def _find_or_rename_sa_widget(
                             break
                         event_type = state.handle_event(event)
 
+                        if event_type == "done":
+                            new_phase = "done"
+                        else:
+                            new_phase = state.compute_phase(has_used_tools)
```
Comment (din0s) [inline EvoScientist/cli/tui_interactive.py]: While testing this PR I've found `has_used_tools` to be a bit of a hit-or-miss. For example when I ask the agent "Can you use the ls tool", I see `Thinking → Working (for a split second) → Researching`, which has to do with when `has_used_tools` is updated vs when `compute_phase` is called.

Would it be possible to derive its value from `state` rather than tracking it in the UI as a parameter? Alternatively, we could move this entire `new_phase` block to the end of the function, so we are sure that the dispatch has updated `has_used_tools`, but I'm not sure if this covers the scenario where the orchestrator doesn't use tools itself other than `task`.

File: EvoScientist/stream/state.py [hunk 2]
```diff
@@ -341,6 +348,45 @@ def handle_event(self, event: dict) -> str:
...
+        """
+        if self.is_thinking:
+            return PHASE_THINKING
+        if has_used_tools and self.has_pending_work():
+            return PHASE_RESEARCHING
+        if self.is_responding:
+            return PHASE_WRITING
+        if has_used_tools:
+            # Tools finished but model hasn't started responding yet
+            return PHASE_WRITING
```
Comment (din0s) [inline EvoScientist/stream/state.py]: Is this the right phase? It could be that the model will decide to do another round of tool calls before emitting the final response.

File: EvoScientist/cli/tui_interactive.py [hunk 3]
```diff
@@ -1384,6 +1382,14 @@ def _find_or_rename_sa_widget(
                             break
                         event_type = state.handle_event(event)
 
+                        if event_type == "done":
+                            new_phase = "done"
```
Comment (din0s) [inline EvoScientist/cli/tui_interactive.py]: Should be `PHASE_DONE`, same with all `self._status_phase` assignments in this file.

File: EvoScientist/cli/tui_interactive.py [hunk 4]
```diff
@@ -2144,6 +2161,7 @@ def _channel_ask_user(ask_user_data: dict) -> dict:
 
             finally:
                 self._busy = False
+                self._status_phase = "idle"
```
Comment (din0s) [inline EvoScientist/cli/tui_interactive.py]: Since we go from "done" to "idle" so fast, it is barely visible in the UI.

File: EvoScientist/stream/state.py [hunk 5]
```diff
@@ -12,6 +12,13 @@
 # These should be excluded from display rendering and "all_done" calculations.
 _INTERNAL_TOOLS = {"ExtractedMemory"}
 
+# Research phase constants used by the TUI status bar.
+PHASE_IDLE = "idle"
+PHASE_THINKING = "thinking"
+PHASE_RESEARCHING = "researching"
+PHASE_WRITING = "writing"
+PHASE_DONE = "done"
```
Comment (din0s) [inline EvoScientist/stream/state.py]: nit but shouldn't these be an enum?

**Review by din0s — COMMENTED** [review]
Nice work, just a few comments/questions regarding edge cases that came up while testing. The highest priority ones are around `compute_phase`. I'm also not entirely sure if the current tests reflect the real order of events that a live session emits.

**Comment by X-iZhang** [comment]
Overall the status bar interaction looks good.

One small thought — the `Done` phase feels a bit redundant. When a turn ends we already render `[Usage: 90,322 in · 475 out · Elapsed: 15s]` underneath, which itself signals completion. The extra green `Done` label kind of duplicates that information, and in practice the Done → idle transition is ~1-2ms so the frame is effectively invisible anyway.

Maybe we could just drop `PHASE_DONE` and go straight from Writing back to idle? Let the status bar focus on in-flight states, and let the Usage line communicate completion. As a side benefit it'd also remove the special case where `compute_phase` never returns `PHASE_DONE`.

**Review by din0s — APPROVED** [review]

**Review by X-iZhang — APPROVED** [review]
